compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/Transforms.cpp

```diff
@@ -17,6 +19,181 @@
+
+    auto dstVecType = cast<VectorType>(mmaOp.getResultType());
+    SmallVector<int64_t, 4> originalSize = *maybeUnrollShape;
+
+    Location loc = mmaOp.getLoc();
+    llvm::MapVector<
+        SmallVector<int64_t>, Value,
+        llvm::DenseMap<SmallVector<int64_t>, unsigned, OffsetMapInfo>>
+        accCache;
+
+    SmallVector<int64_t> loopOrder =
+        getUnrollOrder(mmaOp.getIteratorTypes().size(), mmaOp, options);
+
+    AffineMap lhsPermutationMap = mmaOp.getIndexingMapsArray()[0];
+    AffineMap rhsPermutationMap = mmaOp.getIndexingMapsArray()[1];
+    AffineMap accPermutationMap = mmaOp.getIndexingMapsArray()[2];
+
+    ArrayRef<int64_t> innerAccShape = mmaOp.getAccInnerShape();
+
+    for (SmallVector<int64_t> offsets :
+         StaticTileOffsetRange(originalSize, *targetShape, loopOrder)) {
+      SmallVector<Value> slicesOperands(mmaOp.getNumOperands());
+
+      // Helper to compute the new shape of each operand and extract the slice.
+      auto extractOperand = [&](unsigned index, Value operand,
+                                AffineMap permutationMap,
+                                ArrayRef<int64_t> operandOffets) {
+        SmallVector<int64_t> operandShape = applyPermutationMap(
+            permutationMap, ArrayRef<int64_t>(*targetShape));
+        SmallVector<int64_t> operandStrides(operandOffets.size(), 1);
+        slicesOperands[index] = rewriter.create<vector::ExtractStridedSliceOp>(
+            loc, operand, operandOffets, operandShape, operandStrides);
+      };
+
+      // Extract the new lhs operand.
+      SmallVector<int64_t> lhsOffets =
+          applyPermutationMap(lhsPermutationMap, ArrayRef<int64_t>(offsets));
+      extractOperand(0, mmaOp.getLhs(), lhsPermutationMap, lhsOffets);
+
+      // Extract the new rhs operand.
+      SmallVector<int64_t> rhsOffets =
+          applyPermutationMap(rhsPermutationMap, ArrayRef<int64_t>(offsets));
+      extractOperand(1, mmaOp.getRhs(), rhsPermutationMap, rhsOffets);
+
+      SmallVector<int64_t> accOffets =
+          applyPermutationMap(accPermutationMap, ArrayRef<int64_t>(offsets));
+      // If a version of the accumulator has already been computed, use it
+      // otherwise extract the first version from the original operand.
+      auto *accIt = accCache.find(accOffets);
+      if (accIt != accCache.end()) {
+        slicesOperands[2] = accIt->second;
+      } else {
+        extractOperand(2, mmaOp.getAcc(), accPermutationMap, accOffets);
+      }
+
+      SmallVector<int64_t> dstShape = applyPermutationMap(
+          accPermutationMap, ArrayRef<int64_t>(*targetShape));
+      dstShape.append(innerAccShape.begin(), innerAccShape.end());
+      auto targetType = VectorType::get(dstShape, dstVecType.getElementType());
+
+      // Clone the mma op with the new operands and result type.
+      Operation *newOp =
+          rewriter.create(loc, mmaOp->getName().getIdentifier(), slicesOperands,
+                          targetType, mmaOp->getAttrs());
```

Comment:
Why not use rewriter.create<GPU::MultiMmaOp>? Is it because you want to preserve attributes in op creation? Does `mlir::clone` work?

Reply:
Yes, because it makes it easier to manage the operation attributes. I can try `mlir::clone` and see if that works